packages/components/autocomplete/src/Autocomplete.tsx

```diff
@@ -52,6 +52,13 @@ export interface AutocompleteProps<ItemType>
    * The component will pass the selected "item" as an argument to the function..
    */
   onSelectItem: (item: ItemType) => void;
+
+  /*
+   * Selected item
+   *
+   */
+
+  selectedItem?: ItemType;
```

Comment:
Since we also have the property `defaultValue` should be removed that one explicitly to avoid an overlap?

Reply:
@mshaaban0 added more details now, do you think that its understandable this way?

Reply:
Thank you. I'm still confused about the difference between this and `defaultValue`, maybe it's just me, can you please explain it at least here in comments?

Reply:
Thank you. I'm wondering if it's a good idea to watch `defaultValue` and reflect it in the state instead of introducing another prop?

Reply:
Unfortunately that's not really an option.
The prop defaultValue accepts only a string, while selectedItem accepts everything from ItemType. 
So: defaultValue would be something like "Apple" while the selectedItem is ` {label: "Apple", id: 32445, sys:{foo:[]}}`.
Which part of this item object is the rendered one can be controlled via the itemToString function. 
Also defaultValue is not meant to turn downshift into a controlled input, selectedItem is controlling the selection from outside, which we currently don't allow. We currently can't *unselect* an item at all. We only clear the Textinput.


selectedItem is a property native to downshift, defaultValue is one we introduced on top of it and we are using it in a wrong way.
https://github.com/downshift-js/downshift#selecteditem
https://github.com/downshift-js/downshift#control-props

vs how we use the defaultValue:
https://github.com/downshift-js/downshift#initialinputvalue

Reply:
I'll add an example of how to use this prop to control the selection.

Reply:
If this wasn't mentioned yet, we use this component quite often and in every case, we see the same issues. One is that you can't pass a preselected value as @Lelith pointed out. We tried to workaround this until now by triggering a customer "onChange" event using the reference of the text input field. But that failed for the same reason that we're working with objects instead of strings which in specific cases are not correctly resolved.